### PR TITLE
style: use proper indentation in `ruff.yml`

### PR DESCRIPTION
**Pull request by vil02**
### Describe your change:

[`ruff.yml`](https://github.com/TheAlgorithms/Python/blob/ffb93adf46971be35699e7642d79e90284b3c7f1/.github/workflows/ruff.yml#L14) has some bad indentation. This PR fixes it.

* [x] Fix a bug or typo in an existing workflow

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.

### CODE REVIEW

**Comment by cclauss** [comment]
Let's have machines do this kind of editing so humans can focus on bigger tasks...

Let's add `prettier` or some other yaml formatter to our `.pre-commit-config.yaml` file to make yaml formatting automatic.
* https://github.com/pre-commit/mirrors-prettier#README.md or...
* https://pre-commit.com/hooks.html -- Set `Filter hook type:` to `yaml`

**Comment by vil02** [comment]
I support this idea, although the `yaml` files in this repository do not seem to change very often. It is not that straightforward: formatters remove comments, `yamllint` (with default settings) returns _lots_ of other errors and warnings (which are not that important, in my opinion):
```
./.pre-commit-config.yaml
  1:1       warning  missing document start "---"  (document-start)
  16:8      error    too many spaces after hyphen  (hyphens)
  16:5      error    wrong indentation: expected 6 but found 4  (indentation)
  59:29     warning  too few spaces before comment  (comments)

./.gitpod.yml
  1:1       warning  missing document start "---"  (document-start)

./.github/stale.yml
  4:1       warning  missing document start "---"  (document-start)
  6:81      error    line too long (94 > 80 characters)  (line-length)
  7:81      error    line too long (113 > 80 characters)  (line-length)
  10:81     error    line too long (103 > 80 characters)  (line-length)
  13:81     error    line too long (98 > 80 characters)  (line-length)
  36:81     error    line too long (91 > 80 characters)  (line-length)
  40:81     error    line too long (83 > 80 characters)  (line-length)
  48:81     error    line too long (98 > 80 characters)  (line-length)
  62:81     error    line too long (83 > 80 characters)  (line-length)

./.github/workflows/build.yml
  1:1       warning  missing document start "---"  (document-start)
  3:1       warning  truthy value should be one of [false, true]  (truthy)
  6:25      warning  too few spaces before comment  (comments)

./.github/workflows/project_euler.yml
  1:1       warning  missing document start "---"  (document-start)
  1:1       warning  truthy value should be one of [false, true]  (truthy)
  3:81      error    line too long (88 > 80 characters)  (line-length)
  9:25      warning  too few spaces before comment  (comments)
  25:81     error    line too long (112 > 80 characters)  (line-length)

./.github/workflows/ruff.yml
  2:1       warning  missing document start "---"  (document-start)
  3:1       warning  truthy value should be one of [false, true]  (truthy)

./.github/workflows/directory_writer.yml
  1:81      error    line too long (84 > 80 characters)  (line-length)
  3:1       warning  missing document start "---"  (document-start)
  4:1       warning  truthy value should be one of [false, true]  (truthy)
  19:81     error    line too long (81 > 80 characters)  (line-length)
  20:81     error    line too long (116 > 80 characters)  (line-length)

./.github/ISSUE_TEMPLATE/config.yml
  1:1       warning  missing document start "---"  (document-start)

./.github/ISSUE_TEMPLATE/feature_request.yml
  1:1       warning  missing document start "---"  (document-start)
  8:81      error    line too long (118 > 80 characters)  (line-length)
  9:81      error    line too long (81 > 80 characters)  (line-length)

./.github/ISSUE_TEMPLATE/other.yml
  1:1       warning  missing document start "---"  (document-start)

./.github/ISSUE_TEMPLATE/bug_report.yml
  1:1       warning  missing document start "---"  (document-start)
  8:81      error    line too long (110 > 80 characters)  (line-length)
  46:81     error    line too long (84 > 80 characters)  (line-length)

```

I can have a look at this problem. However, it seems to be out of scope for this PR.

**Comment by cclauss** [comment]
Linter != formatter.  Let’s focus on prettier, not yamllint.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -59,3 +59,9 @@ repos:
           - --install-types # See mirrors-mypy README.md
           - --non-interactive
         additional_dependencies: [types-requests]
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: 'v3.1.0'
+    hooks:
+      - id: prettier
+        types: [yaml]
```
Comment (cclauss) [inline .pre-commit-config.yaml:67]:
```suggestion
        types: [toml, yaml]
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -59,3 +59,9 @@ repos:
           - --install-types # See mirrors-mypy README.md
           - --non-interactive
         additional_dependencies: [types-requests]
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: 'v3.1.0'
+    hooks:
+      - id: prettier
+        types: [toml, yaml]
```
Comment (cclauss) [inline .pre-commit-config.yaml:67]:
```suggestion
        types_or: [toml, yaml]
```

**Comment by vil02** [comment]
@cclauss Please let me know when you are done...

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -59,3 +59,13 @@ repos:
...
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: "v3.1.0"
+    hooks:
+      - id: prettier
+        types_or: [toml, yaml]
+        exclude: |
+          (?x)^(
+              \.github/workflows/(build|directory_writer|project_euler|ruff).yml
+          )$
```
Comment (vil02) [inline .pre-commit-config.yaml:71]: So actually this would not create the change from 51a75c514816ac1c63c01b9f4b69ba3a1ad81835.

**Comment by cclauss** [comment]
Can you remove the exclude on 68-71 and then run the following on your local machine?
% `pre-commit install`
% `pre-commit run --all-files`

The problem with running this on GitHub Actions is that it does not allow changes to be made to GHA files.

**Review by cclauss — APPROVED** [review]
Nice!

**Pull request merged**